sys/include/sema.h

```diff
@@ -69,57 +89,25 @@ int sema_create(sema_t *sema, unsigned int value);
  *          The Open Group Base Specifications Issue 7, sem_destroy()
  *      </a>
  *
+ * Destroying a semaphore upon which other threads are currently blocked
+ * will wake the other threads causing the @ref sema_wait (or
+ * @ref sema_wait_timed) to return error (-1).
+ *
  * @param[in] sema  The semaphore to destroy.
  *
  * @return  0 on success.
- * @return  -EINVAL, if semaphore is invalid.
  */
 int sema_destroy(sema_t *sema);
 
 /**
  * @brief   Wait for a semaphore being posted.
  *
- * @pre Message queue of active thread is initialized (see @ref msg_init_queue()).
- *
- * @param[in]  sema     A semaphore.
- * @param[in]  timeout  Time in microseconds until the semaphore times out. 0 for no timeout.
- * @param[out] msg      Container for a spurious message during the timed wait (result == -EAGAIN).
- *
- * @return  0 on success
- * @return  -EINVAL, if semaphore is invalid.
- * @return  -ETIMEDOUT, if the semaphore times out.
- * @return  -ECANCELED, if the semaphore was destroyed.
- * @return  -EAGAIN, if the thread received a message while waiting for the lock.
- */
-int sema_wait_timed_msg(sema_t *sema, uint64_t timeout, msg_t *msg);
```

Comment:
No, lwIP was calling `sema_wait_timed` which was calling `sema_wait_timed_msg`. However, your next point about the return values makes that lwIP requires fixing, I'll rollback the return values to the original ones.